packages/camera/camera_avfoundation/ios/camera_avfoundation/Sources/camera_avfoundation/CameraPlugin.swift

```diff
@@ -238,25 +240,27 @@ extension CameraPlugin: FCPCameraApi {
       mediaSettings: settings,
       mediaSettingsWrapper: mediaSettingsAVWrapper,
       captureDeviceFactory: captureDeviceFactory,
+      audioCaptureDeviceFactory: {
+        FLTDefaultCaptureDevice(device: AVCaptureDevice.default(for: .audio)!)
+      },
       captureSessionFactory: captureSessionFactory,
       captureSessionQueue: captureSessionQueue,
       captureDeviceInputFactory: captureDeviceInputFactory,
       initialCameraName: name
     )
 
-    var error: NSError?
-    let newCamera = FLTCam(configuration: camConfiguration, error: &error)
+    do {
+      let newCamera = try FLTDefaultCam(configuration: camConfiguration)
 
-    if let error = error {
-      completion(nil, CameraPlugin.flutterErrorFromNSError(error))
-    } else {
       camera?.close()
       camera = newCamera
 
       FLTEnsureToRunOnMainQueue { [weak self] in
         guard let strongSelf = self else { return }
         completion(NSNumber(value: strongSelf.registry.register(newCamera)), nil)
       }
+    } catch let error as NSError {
```

Comment:
Was the error originated from our own code or from AVFoundation?

Reply:
Both:
1. `AVCaptureDeviceInput deviceInputWithDevice` (AVFoundation)
2. `AVCaptureDevice lockForConfiguration` (AVFoundation)
3. `FLTDefaultCam.setCaptureSessionPreset` (our own code)

IMO the whole package deserves a clean up of errors in the future because there is absolutely no consistency in the error format